aether-kernel/aether/kernel/api/project_artefacts.py

```diff
@@ -134,10 +136,24 @@ def upsert_project_artefacts(
     # 4. create/update the mappings
     for raw_mapping in mappings:
         ignore_fields = ['name']
-        if raw_mapping.get('definition') is None:
+        mapping_definition = raw_mapping.get('definition')
+        if mapping_definition is None:
             # in case of no mapping rules were indicated, do not update them.
             ignore_fields.append('definition')
-
+            mapping_definition = {'mappings': [], 'entities': {}}
+        else:
```

Comment:
This else is making CoudhDB-Sync test fail, could you calculate the `entities` property only if doesn't come in the mapping definition?

Reply:
`else if 'entities' not in mapping_definition:`

Reply:
Take in mind that ODK and CouchDB-Sync provide the whole definition, rules and entities.